### PR TITLE
feat: make cookies have sameSite key by default

### LINKED ISSUE #6892: Make adding `sameSite` value to objects yielded by `cy.getCookie/setCookie/getCookies` the default
Remove the `experimentalGetCookiesSameSite` flag and make this behavior the default for all Cypress users.

### PR DESCRIPTION
**Pull request by flotwig**
BREAKING CHANGE: modifies the shape of Cookie objects

- Closes #6892

### User facing changelog

- Values yielded by `cy.setCookie`, `cy.getCookie`, and `cy.getCookies` will now always contain a `sameSite` attribute.
- Removed `experimentalGetCookiesSameSite` configuration flag, since this behavior is now the default.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/2918
- [x] Have API changes been updated in the [`type definitions`](../cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](../cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by flotwig** [comment]
I think that since users with `experimentalGetCookiesSameSite` enabled will not experience a change in behavior, we don't need to add a deprecation warning at runtime. The migration guide will presumably document that this config option can be removed, and we should probably comment on the closed issues that this is no longer "experimental", and the flag is no longer necessary.

**Comment by brian-mann** [comment]
I think in this case that's probably fine, but since we're doing the `experimental*` pattern more frequently now it makes me wonder what our best practice should be. I'm actually on the side that we should hard error and throw and tell them this option has been removed.

I think that approach is much better than to silently pass because anyone looking at the code later will be sufficiently confused - won't be able to find the option anywhere in the docs, and will wonder if this is actually taking effect anywhere. Throwing forces the user to update their code, but it's the cleanest way to communicate the intention. Additionally, imagine if we had a situation where the options of the `experimental*` feature change to have new official meaning when it gets released. 

I can think of a few other scenarios but those are the ones off the top of my head. I believe we already have logic in `config` to notify the user of renamed / removed properties. I'd also be okay with a deprecation notice, but unless people literally read their `stdout` they won't see it.

**Comment by flotwig** [comment]
I guess it's either make the user pay the cost now, by forcing them to update their config, or pay the cost later, by potentially confusing users that have not read the migration guide/seen the issue comment telling them to remove the config option.

Adding a deprecation notice to `stdout` sounds like a decent compromise between the two that will help make the 5.x upgrade easier, so that's what I'll do for this PR.

**Comment by brian-mann** [comment]
Okay cool. Use the existing config warning/deprecation machinery.

File: packages/server/lib/errors.js [hunk 1]
```diff
@@ -923,6 +923,11 @@ const getMsgByType = function (type, arg1 = {}, arg2, arg3) {
         Enable write permissions to this directory to ensure screenshots and videos are stored.
 
         If you don't require screenshots or videos to be stored you can safely ignore this warning.`
+    case 'EXPERIMENTAL_SAMESITE_REMOVED':
+      return stripIndent`\
+        The \`experimentalGetCookiesSameSite\` configuration option was removed in Cypress 5. Yielding the \`sameSite\` property is now the default behavior of the \`cy.cookie\` commands.
```
Comment (brian-mann) [inline packages/server/lib/errors.js:928]: I think the convention when listing versions is to say `in Cypress version x.x.x`

Comment (flotwig) [reply packages/server/lib/errors.js:928]: This was the only prior art I could find, which also isn't that:

https://github.com/cypress-io/cypress/blob/c00f6cdac2fddaeeabb8ee4bd4c62adbbfd8ea04/packages/server/lib/errors.js#L140-L141

Comment (jennifer-shehane) [reply packages/server/lib/errors.js:928]: In our migration guide we usually say Cypress 5, so as not to be restricted of it only applying to Cypress 5.0.0 and not to 5.1.0, etc.

**Pull request merged**